cmd/kube-controller-manager/app/plugins.go

```diff
@@ -43,7 +43,22 @@ func ProbeRecyclableVolumePlugins() []volume.VolumePlugin {
 	// The list of plugins to probe is decided by the kubelet binary, not
 	// by dynamic linking or other "magic".  Plugins will be analyzed and
 	// initialized later.
-	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins()...)
-	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins(defaultRecycler(60))...)
```

Comment:
Hrm - I think we want to be able to rip out the entire process of starting the container - we will have specific security constraints, rules, labels, node selectors, etc.  I don't think it's sufficient to just change image name.

I think a volume plugin is mostly orthogonal to a recycler implementation - I don't want them to be coupled, and I do think the responsibilities need to be distinct.

----- Original Message -----
> > @@ -43,7 +43,22 @@ func ProbeRecyclableVolumePlugins()
> > []volume.VolumePlugin {
> >  	// The list of plugins to probe is decided by the kubelet binary, not
> >  	// by dynamic linking or other "magic".  Plugins will be analyzed and
> >  	// initialized later.
> > -	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins()...)
> > -	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins()...)
> > +	allPlugins = append(allPlugins,
> > host_path.ProbeVolumePlugins(defaultRecycler(60))...)
> 
> Agree to all three.  This was hastily thrown together to avoid the hard-coded
> busybox requirement.
> 
> Is it worthwhile to override how something gets scrubbed?  I can make a
> custom kube image for the scrubber.  Do we need more than that?
> 
> It might be we just need to override image name.  I can figure out X seconds
> timeout per gig to scrub in a volume, which removes these magic numbers.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9870/files#r32736327